src/java.base/share/classes/jdk/internal/loader/URLClassPath.java

```diff
@@ -218,7 +217,7 @@ public URLClassPath(URL[] urls, AccessControlContext acc) {
         if (closed) {
             return Collections.emptyList();
         }
-        List<IOException> result = new LinkedList<>();
+        List<IOException> result = new ArrayList<>();
```

Comment:
We'd better be cautious about this replacement since its [caller](https://github.com/openjdk/jdk/blob/73029fe10a8a814a8c5f5221f2e667fd14a5b379/src/java.base/share/classes/java/net/URLClassLoader.java#L363) will remove the first element of this array, that's one of the scenarios where LinkedList usually has better performance than ArrayList.

Just IMHO, I suggest replacing them only if there is a performance improvement. Change fields type will break users' existing application code, they might reflectively modify these values.

Reply:
If that's the only use case, I recommend changing the return type to a deque, and replace the linked list with an array deque instead (as done elsewhere in this pr)

Reply:
Looks like it's never specified in JavaDoc which particular implementation of List is used in fields of affected classes, so it's quite odd to me that someone would rely on that when using reflection. But your point about backward compatibility is reasonable, so I'll revert mentioned changes.

Reply:
Nothing outside of the JDK should be hacking into this private field of a non-exposed class, this should not be a concern here.

Reply:
@AlanBateman so is it ok to keep `ArrayLists`?

Reply:
> @AlanBateman so is it ok to keep `ArrayLists`?

One thing to look out for is usages of 2-arg add method that inserts at a specific position. This shouldn't be a concern in URLClassPath.closeLoaders (assuming this is this usage where the question arises).

Reply:
@AlanBateman looks like `List.add(o, i)` is not used at all in scope of these changes.